central/imageintegration/store/defaults.go

```diff
@@ -112,6 +113,21 @@ var DefaultImageIntegrations = []*storage.ImageIntegration{
 	},
 }
 
+// DefaultScannerV4Integration is the default Scanner V4 integration.
+var DefaultScannerV4Integration = &storage.ImageIntegration{
+	Id:   "a87471e6-9678-4e66-8348-91e302b6de07",
+	Name: "ScannerV4",
+	Type: scannerv4.TypeString,
```

Comment:
More of a general comment: it'd be great in some future to also re-use the pattern of `Traits` here as well which has been defined in the process of adding declarative config (here is [the access scope as an example](https://github.com/stackrox/stackrox/blob/master/central/role/default.go#L22C1-L24)), this way then the datastore itself would just verify based on the traits and we wouldn't have this special casing based on all the different types.

Reply:
Will reach out to chat more about this, took a quick look at the existing traits and some initial thoughts: this new scanner integration is part default and part mutable, the older scanner integration is fully mutable and a default - both defaults are conditional on aspects of the installation (ScannerV2 polls the scanner svc endpoint indefinitely prior to creation, and ScannerV4 relies on an environment variable to be set at install time) - they may not fit the current traits framework if I understand it correctly - we may have to add the ability to specify per field (instead of per object) traits which could be unnecessary complexity if this is an edge case.